.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,338 @@
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Install tools
+        run: make install-tools
+      - name: Upload tool binaries
+        uses: actions/upload-artifact@v2
+        with:
+          name: tool-binaries
+          path: /home/runner/go/bin
+  lint:
+    runs-on: ubuntu-latest
+    needs: [setup-environment]
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Setup env
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: Download tool binaries
+        uses: actions/download-artifact@v2
+        with:
+          name: tool-binaries
+          path: /home/runner/go/bin
+      - name: Add execute permissions to tool binaries
+        run: chmod -R +x /home/runner/go/bin
+      - name: Lint
+        run: make -j4 gochecklicense goimpi golint gomisspell
+      - name: Gen Metadata
+        run: make genmdata
+      - name: Check generated metadata
+        run: git diff --exit-code || (echo 'Generated code is out of date, please run "make genmdata" and commit the changes in this PR.' && exit 1)
+  cross-compile:
+    runs-on: ubuntu-latest
+    needs: [setup-environment]
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
+      - name: Setup env
+        run: |
+          echo "GOPATH=$(go env GOPATH)" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+      - name: Restore module cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-go-modules
+        with:
+          path: /home/runner/go/pkg/mod
+          key: go-pkg-mod-${{ runner.os }}-${{ hashFiles('./go.mod') }}
+      - name: Download tool binaries
+        uses: actions/download-artifact@v2
+        with:
+          name: tool-binaries
+          path: /home/runner/go/bin
+      - name: Add execute permissions to tool binaries
+        run: chmod -R +x /home/runner/go/bin
+      - name: Build collector for all archs
+        run: grep ^binaries-all-sys Makefile|fmt -w 1|tail -n +2|xargs make
+      - name: Upload collector binaries
+        uses: actions/upload-artifact@v2
+        with:
+          name: collector-binaries
+          path: ./bin
+  loadtestpre:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.splitloadtest.outputs.matrix }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
```

Comment:
`loadtestpre` does some pre-processing so we can simulate CircleCI testsharding on GitHub Actions. This cuts down the runtime of loadtests from 15mins to ~5mins
This job does the following: 

1. Get a list of all loadtest tests using `make testbed-list-loadtest`
2. Split tests into groups of two (`TestIdleMode|TestBallastMemory`, `TestLog10kDPS|TestMetric10kDPS` etc)
3. Create a JSON containing each test grouping. This JSON will be consumed by the [strategy.matrix](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix) of the loadtest job using [`fromJSON()`](https://github.blog/changelog/2020-04-15-github-actions-new-workflow-features/). The strategy.matrix is how we get to run the loadtests in parallel